doc/concepts/simulation.rst

```diff
@@ -2,28 +2,33 @@
 
 Simulations
 ===========
-A simulation is the executable form of a model and is used to interact with and monitor the model state. In the simulation the neuron model is initiated and the spike exchange and the integration for each cell group are scheduled.
+A simulation is the executable form of a model and is used to interact with and monitor the model state.
+In the simulation the neuron model is initiated and the spike exchange and the integration for each cell
+group are scheduled.
 
 From recipe to simulation
 -------------------------
 
 To build a simulation the following are needed:
 
-    * A recipe that describes the cells and connections in the model.
-    * A context used to execute the simulation.
+    * A :ref:`recipe <modelrecipe>` that describes the cells and connections in the model.
+    * A :ref:`domain decomposition <modeldomdec>` that describes the distribution of the
+      model over the local and distributed :ref:`hardware resources <modelhardware>`.
+    * An :ref:`execution context <modelcontext>` used to execute the simulation.
 
-The workflow to build a simulation is to first generate a domain decomposition that describes the distribution of the model over the local and distributed hardware resources (see :ref:`modeldomdec`), then build the simulation from the recipe, the domain decomposition and the execution context. Optionally experimental inputs  that can change between model runs, such as external spike trains, can be injected.
-
-The recipe describes the model, the domain decomposition describes how the cells in the model are assigned to hardware resources and the context is used to execute the simulation.
+Optionally, experimental inputs that can change between model runs, such as external spike trains, can be injected.
```

Comment:
I can't find a definition for 'injecting', should we link to it? Or to how one might do this?

Reply:
I think this is not too important of a concept to be explained in the concepts page. It is part of the simulation [cpp API](https://arbor.readthedocs.io/en/latest/cpp/simulation.html#_CPPv4N3arb10simulation13inject_eventsERK10pse_vector), but it doesn't look like it's exposed in python yet.

Reply:
If you think it's too ambiguous I can modify this or mention that it's only exposed in the C++ API (with a link)

Reply:
It's Arbor-specific jargon, so some idea of what it means or where to look for details would be nice. It's covered in the C++ API page, but an problem in the Python API page is revealed: a binning class is defined and the binning policy method of `simulation` is annotated, but any mention of event injection is missing. The Python reference ( :smiley: ) reveals that quite a bit is exposed though! `arbor.event_generator` exists.

The C++ API page does not _explain_ events and injection either by the way. So maybe this is the right place?

Reply:
I'm just going to remove the line. It doesn't really add much to the description here.

Reply:
We could go into detail about this specific concept here, but it'll only add confusion as it is not an essential feature in arbor, that's probably also why it hasn't been exposed in python yet!

Reply:
So it is already exposed, at least in part (I don't know how much of a part).

I would like to keep some trace of this, because if it is implemented it should be documented and it should be here, eventually. What's a good way of keeping track of things like this? We've collected some todos in the Slack, I've collected some in an Issue before, Sphinx has the `..todo::` directive, we could file separate Issues for each thing we identify. They're all options as far as I'm concerned, but maybe we should choose one way of tracking TODOs. What has your preference?

Reply:
I don't think spike injection needs to appear in the concept docs at all. Not every part of the API is a concept after all. Spike injection is documented in the C++ API and when it is exposed to the python API it will appear there.
As for keeping track of todos, I would rather not have them in the docs unless they are important concepts/parts of the API that have been implemented but not yet documented. We can have a separate issue for each thing we identify, or one issue with a group of small fixes.

Reply:
OK, not here. Keeping track: I made [this](https://mcnest.slack.com/archives/C019BVBKS5V/p1604484691003200) post in the slack, is that enough or shall I make an Issue out of it?